File: SoftUni - C# OOP/Polymorphism/Polymorphism - Lab/02. Animals/Animal.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace Animals
{
    public class Animal
    {
        public string Name { get; set; }
        public string Food { get; set; }
        public Animal (string name,string food)
        {
            this.Name = name;
            this.Food = food;
        }
        public virtual string ExplainSelf()
        {
            return $"I am {this.Name} and my fovourite food is {this.Food}";
        }
    }
}

File: SoftUni - Java OOP/Polymorphism/Polymorphism - Lab/_03_Animals/Animal.java
package _03_Animals;

public abstract class Animal {
    private String name;
    private String favouriteFood;
    public Animal(String name, String favouriteFood) {
        this.name = name;
        this.favouriteFood = favouriteFood;
    }
    public String explainSelf() {
        return "I am %s and my favourite food is %s ".formatted(this.name, this.favouriteFood);
    }

    public String getName() {
        return name;
    }

    public String getFavouriteFood() {
        return favouriteFood;
    }
}
